In [1]:
from IPython.display import Image
import numpy as np
import pandas as pd
import seaborn as sns
import warnings

# warning 무시
warnings.filterwarnings('ignore')

# e notation 표현 방식 변경
pd.options.display.float_format = '{:.2f}'.format

# 모든 컬럼 표시
pd.set_option('display.max_columns', None)

In [2]:
df = sns.load_dataset('titanic')
df

survived  pclass     sex   age  sibsp  parch  fare embarked   class  \
0           0       3    male 22.00      1      0  7.25        S   Third   
1           1       1  female 38.00      1      0 71.28        C   First   
2           1       3  female 26.00      0      0  7.92        S   Third   
3           1       1  female 35.00      1      0 53.10        S   First   
4           0       3    male 35.00      0      0  8.05        S   Third   
..        ...     ...     ...   ...    ...    ...   ...      ...     ...   
886         0       2    male 27.00      0      0 13.00        S  Second   
887         1       1  female 19.00      0      0 30.00        S   First   
888         0       3  female   NaN      1      2 23.45        S   Third   
889         1       1    male 26.00      0      0 30.00        C   First   
890         0       3    male 32.00      0      0  7.75        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

**컬럼(columns) 설명**

- survivied: 생존여부 (1: 생존, 0: 사망)
- pclass: 좌석 등급 (1등급, 2등급, 3등급)
- sex: 성별
- age: 나이
- sibsp: 형제 + 배우자 수
- parch: 부모 + 자녀 수
- fare: 좌석 요금
- embarked: 탑승 항구 (S, C, Q)
- class: pclass와 동일
- who: 남자(man), 여자(woman), 아이(child)
- adult_male: 성인 남자 여부
- deck: 데크 번호 (알파벳 + 숫자 혼용)
- embark_town: 탑승 항구 이름
- alive: 생존여부 (yes, no)
- alone: 혼자 탑승 여부

## 새로운 칼럼 추가

In [3]:
df1 = df.copy()
df1.head()

survived  pclass     sex   age  sibsp  parch  fare embarked  class    who  \
0         0       3    male 22.00      1      0  7.25        S  Third    man   
1         1       1  female 38.00      1      0 71.28        C  First  woman   
2         1       3  female 26.00      0      0  7.92        S  Third  woman   
3         1       1  female 35.00      1      0 53.10        S  First  woman   
4         0       3    male 35.00      0      0  8.05        S  Third    man   

   adult_male deck  embark_town alive  alone  
0        True  NaN  Southampton    no  False  
1       False    C    Cherbourg   yes  False  
2       False  NaN  Southampton   yes   True  
3       False    C  Southampton   yes  False  
4        True  NaN  Southampton    no   True

In [4]:
df1['VIP'] = True

In [5]:
df1.head()

survived  pclass     sex   age  sibsp  parch  fare embarked  class    who  \
0         0       3    male 22.00      1      0  7.25        S  Third    man   
1         1       1  female 38.00      1      0 71.28        C  First  woman   
2         1       3  female 26.00      0      0  7.92        S  Third  woman   
3         1       1  female 35.00      1      0 53.10        S  First  woman   
4         0       3    male 35.00      0      0  8.05        S  Third    man   

   adult_male deck  embark_town alive  alone   VIP  
0        True  NaN  Southampton    no  False  True  
1       False    C    Cherbourg   yes  False  True  
2       False  NaN  Southampton   yes   True  True  
3       False    C  Southampton   yes  False  True  
4        True  NaN  Southampton    no   True  True

In [6]:
df1.insert(5, 'RICH', df1['fare'] > 100)

In [7]:
df1

survived  pclass     sex   age  sibsp   RICH  parch  fare embarked  \
0           0       3    male 22.00      1  False      0  7.25        S   
1           1       1  female 38.00      1  False      0 71.28        C   
2           1       3  female 26.00      0  False      0  7.92        S   
3           1       1  female 35.00      1  False      0 53.10        S   
4           0       3    male 35.00      0  False      0  8.05        S   
..        ...     ...     ...   ...    ...    ...    ...   ...      ...   
886         0       2    male 27.00      0  False      0 13.00        S   
887         1       1  female 19.00      0  False      0 30.00        S   
888         0       3  female   NaN      1  False      2 23.45        S   
889         1       1    male 26.00      0  False      0 30.00        C   
890         0       3    male 32.00      0  False      0  7.75        Q   

      class    who  adult_male deck  embark_town alive  alone   VIP  
0     Third    man        True  NaN  Southampton    no  False  True  
1     First  woman       False    C    Cherbourg   yes  False  True  
2     Third  woman       False  NaN  Southampton   yes   True  True  
3     First  woman       False    C  Southampton   yes  False  True  
4     Third    man        True  NaN  Southampton    no   True  True  
..      ...    ...         ...  ...          ...   ...    ...   ...  
886  Second    man        True  NaN  Southampton    no   True  True  
887   First  woman       False    B  Southampton   yes   True  True  
888   Third  woman       False  NaN  Southampton    no  False  True  
889   First    man        True    C    Cherbourg   yes   True  True  
890   Third    man        True  NaN   Queenstown    no   True  True  

[891 rows x 17 columns]

## 삭제

삭제는 **행(row) 삭제와 열(column) 삭제**로 구분할 수 있습니다.

### 행 (row) 삭제
행 삭제시 **index를 지정하여 삭제**합니다.

In [8]:
df1.drop(1)

survived  pclass     sex   age  sibsp   RICH  parch  fare embarked  \
0           0       3    male 22.00      1  False      0  7.25        S   
2           1       3  female 26.00      0  False      0  7.92        S   
3           1       1  female 35.00      1  False      0 53.10        S   
4           0       3    male 35.00      0  False      0  8.05        S   
5           0       3    male   NaN      0  False      0  8.46        Q   
..        ...     ...     ...   ...    ...    ...    ...   ...      ...   
886         0       2    male 27.00      0  False      0 13.00        S   
887         1       1  female 19.00      0  False      0 30.00        S   
888         0       3  female   NaN      1  False      2 23.45        S   
889         1       1    male 26.00      0  False      0 30.00        C   
890         0       3    male 32.00      0  False      0  7.75        Q   

      class    who  adult_male deck  embark_town alive  alone   VIP  
0     Third    man        True  NaN  Southampton    no  False  True  
2     Third  woman       False  NaN  Southampton   yes   True  True  
3     First  woman       False    C  Southampton   yes  False  True  
4     Third    man        True  NaN  Southampton    no   True  True  
5     Third    man        True  NaN   Queenstown    no   True  True  
..      ...    ...         ...  ...          ...   ...    ...   ...  
886  Second    man        True  NaN  Southampton    no   True  True  
887   First  woman       False    B  Southampton   yes   True  True  
888   Third  woman       False  NaN  Southampton    no  False  True  
889   First    man        True    C    Cherbourg   yes   True  True  
890   Third    man        True  NaN   Queenstown    no   True  True  

[890 rows x 17 columns]

행 삭제시 **범위를 지정하여 삭제**할 수 있습니다.

In [9]:
df1.drop(np.arange(10))

survived  pclass     sex   age  sibsp   RICH  parch  fare embarked  \
10          1       3  female  4.00      1  False      1 16.70        S   
11          1       1  female 58.00      0  False      0 26.55        S   
12          0       3    male 20.00      0  False      0  8.05        S   
13          0       3    male 39.00      1  False      5 31.27        S   
14          0       3  female 14.00      0  False      0  7.85        S   
..        ...     ...     ...   ...    ...    ...    ...   ...      ...   
886         0       2    male 27.00      0  False      0 13.00        S   
887         1       1  female 19.00      0  False      0 30.00        S   
888         0       3  female   NaN      1  False      2 23.45        S   
889         1       1    male 26.00      0  False      0 30.00        C   
890         0       3    male 32.00      0  False      0  7.75        Q   

      class    who  adult_male deck  embark_town alive  alone   VIP  
10    Third  child       False    G  Southampton   yes  False  True  
11    First  woman       False    C  Southampton   yes   True  True  
12    Third    man        True  NaN  Southampton    no   True  True  
13    Third    man        True  NaN  Southampton    no  False  True  
14    Third  child       False  NaN  Southampton    no   True  True  
..      ...    ...         ...  ...          ...   ...    ...   ...  
886  Second    man        True  NaN  Southampton    no   True  True  
887   First  woman       False    B  Southampton   yes   True  True  
888   Third  woman       False  NaN  Southampton    no  False  True  
889   First    man        True    C    Cherbourg   yes   True  True  
890   Third    man        True  NaN   Queenstown    no   True  True  

[881 rows x 17 columns]

**fancy indexing**을 활용하여 삭제할 수 있습니다.

In [10]:
df1.drop([1, 3, 5, 7, 9])

survived  pclass     sex   age  sibsp   RICH  parch  fare embarked  \
0           0       3    male 22.00      1  False      0  7.25        S   
2           1       3  female 26.00      0  False      0  7.92        S   
4           0       3    male 35.00      0  False      0  8.05        S   
6           0       1    male 54.00      0  False      0 51.86        S   
8           1       3  female 27.00      0  False      2 11.13        S   
..        ...     ...     ...   ...    ...    ...    ...   ...      ...   
886         0       2    male 27.00      0  False      0 13.00        S   
887         1       1  female 19.00      0  False      0 30.00        S   
888         0       3  female   NaN      1  False      2 23.45        S   
889         1       1    male 26.00      0  False      0 30.00        C   
890         0       3    male 32.00      0  False      0  7.75        Q   

      class    who  adult_male deck  embark_town alive  alone   VIP  
0     Third    man        True  NaN  Southampton    no  False  True  
2     Third  woman       False  NaN  Southampton   yes   True  True  
4     Third    man        True  NaN  Southampton    no   True  True  
6     First    man        True    E  Southampton    no   True  True  
8     Third  woman       False  NaN  Southampton   yes  False  True  
..      ...    ...         ...  ...          ...   ...    ...   ...  
886  Second    man        True  NaN  Southampton    no   True  True  
887   First  woman       False    B  Southampton   yes   True  True  
888   Third  woman       False  NaN  Southampton    no  False  True  
889   First    man        True    C    Cherbourg   yes   True  True  
890   Third    man        True  NaN   Queenstown    no   True  True  

[886 rows x 17 columns]

### 열 (column) 삭제

In [11]:
df1.head()

survived  pclass     sex   age  sibsp   RICH  parch  fare embarked  class  \
0         0       3    male 22.00      1  False      0  7.25        S  Third   
1         1       1  female 38.00      1  False      0 71.28        C  First   
2         1       3  female 26.00      0  False      0  7.92        S  Third   
3         1       1  female 35.00      1  False      0 53.10        S  First   
4         0       3    male 35.00      0  False      0  8.05        S  Third   

     who  adult_male deck  embark_town alive  alone   VIP  
0    man        True  NaN  Southampton    no  False  True  
1  woman       False    C    Cherbourg   yes  False  True  
2  woman       False  NaN  Southampton   yes   True  True  
3  woman       False    C  Southampton   yes  False  True  
4    man        True  NaN  Southampton    no   True  True

열 삭제시 **반드시 `axis=1` 옵션을 지정**해야 합니다. 2번째 위치에 지정시 `axis=`을 생략할 수 있습니다.

In [12]:
df1.drop('class', axis=1).head()

survived  pclass     sex   age  sibsp   RICH  parch  fare embarked    who  \
0         0       3    male 22.00      1  False      0  7.25        S    man   
1         1       1  female 38.00      1  False      0 71.28        C  woman   
2         1       3  female 26.00      0  False      0  7.92        S  woman   
3         1       1  female 35.00      1  False      0 53.10        S  woman   
4         0       3    male 35.00      0  False      0  8.05        S    man   

   adult_male deck  embark_town alive  alone   VIP  
0        True  NaN  Southampton    no  False  True  
1       False    C    Cherbourg   yes  False  True  
2       False  NaN  Southampton   yes   True  True  
3       False    C  Southampton   yes  False  True  
4        True  NaN  Southampton    no   True  True

**다수의 컬럼(column) 삭제**도 가능합니다.

In [13]:
df1.drop(['who', 'deck', 'alive'], axis=1)

survived  pclass     sex   age  sibsp   RICH  parch  fare embarked  \
0           0       3    male 22.00      1  False      0  7.25        S   
1           1       1  female 38.00      1  False      0 71.28        C   
2           1       3  female 26.00      0  False      0  7.92        S   
3           1       1  female 35.00      1  False      0 53.10        S   
4           0       3    male 35.00      0  False      0  8.05        S   
..        ...     ...     ...   ...    ...    ...    ...   ...      ...   
886         0       2    male 27.00      0  False      0 13.00        S   
887         1       1  female 19.00      0  False      0 30.00        S   
888         0       3  female   NaN      1  False      2 23.45        S   
889         1       1    male 26.00      0  False      0 30.00        C   
890         0       3    male 32.00      0  False      0  7.75        Q   

      class  adult_male  embark_town  alone   VIP  
0     Third        True  Southampton  False  True  
1     First       False    Cherbourg  False  True  
2     Third       False  Southampton   True  True  
3     First       False  Southampton  False  True  
4     Third        True  Southampton   True  True  
..      ...         ...          ...    ...   ...  
886  Second        True  Southampton   True  True  
887   First       False  Southampton   True  True  
888   Third       False  Southampton  False  True  
889   First        True    Cherbourg   True  True  
890   Third        True   Queenstown   True  True  

[891 rows x 14 columns]

삭제된 내용을 바로 적용하려면 
1. `inplace=True`를 지정합니다.
2. 변수에 **재대입** 하여 결과를 반영합니다.

## 연습문제

- `df1`에서 1, 3, 5번행을 삭제해 주세요
- `df1`에서 `embarked`, `class`, `alone` 컬럼을 삭제해 주세요
- `df1`의 상위 10개 행을 출력하세요

In [ ]:
df1 = df.copy()

## 컬럼간 연산

**컬럼(column) 과 컬럼 사이의 연산을 매우 쉽게 적용**할 수 있습니다.

In [14]:
df1 = df.copy()

**family(가족)** 의 총합은 **sibsp**컬럼과 **parch**의 합산으로 구할 수 있습니다.

In [15]:
df1['family'] = df1['sibsp'] + df1['parch']

In [16]:
df1.head()

survived  pclass     sex   age  sibsp  parch  fare embarked  class    who  \
0         0       3    male 22.00      1      0  7.25        S  Third    man   
1         1       1  female 38.00      1      0 71.28        C  First  woman   
2         1       3  female 26.00      0      0  7.92        S  Third  woman   
3         1       1  female 35.00      1      0 53.10        S  First  woman   
4         0       3    male 35.00      0      0  8.05        S  Third    man   

   adult_male deck  embark_town alive  alone  family  
0        True  NaN  Southampton    no  False       1  
1       False    C    Cherbourg   yes  False       1  
2       False  NaN  Southampton   yes   True       0  
3       False    C  Southampton   yes  False       1  
4        True  NaN  Southampton    no   True       0

**문자열의 합 (이어붙히기)도 가능**합니다.

In [17]:
df1['gender'] = df1['who'] + '-' + df1['sex']

In [18]:
df1.head()

survived  pclass     sex   age  sibsp  parch  fare embarked  class    who  \
0         0       3    male 22.00      1      0  7.25        S  Third    man   
1         1       1  female 38.00      1      0 71.28        C  First  woman   
2         1       3  female 26.00      0      0  7.92        S  Third  woman   
3         1       1  female 35.00      1      0 53.10        S  First  woman   
4         0       3    male 35.00      0      0  8.05        S  Third    man   

   adult_male deck  embark_town alive  alone  family        gender  
0        True  NaN  Southampton    no  False       1      man-male  
1       False    C    Cherbourg   yes  False       1  woman-female  
2       False  NaN  Southampton   yes   True       0  woman-female  
3       False    C  Southampton   yes  False       1  woman-female  
4        True  NaN  Southampton    no   True       0      man-male

연산시 1개의 컬럼이라도 **NaN 값을 포함하고 있다면 결과는 NaN** 이 됩니다.

In [20]:
df1['round'] = round(df1['fare'] / df1['age'], 2)
df1.loc[df1['age'].isnull(), ['fare', 'age', 'round']].head()

fare  age  round
5   8.46  NaN    NaN
17 13.00  NaN    NaN
19  7.22  NaN    NaN
26  7.22  NaN    NaN
28  7.88  NaN    NaN

### 연습문제

`iris` 붓꽃 데이터셋을 활용하여 다음의 문제를 풀어주세요

<img src="https://machinelearninghd.com/wp-content/uploads/2021/03/iris-dataset.png"
     align="left" 
     width="500">

(참고)
- species: 붓꽃 데이터의 종류
- `sepal_length`: 꽃받침의 길이
- `sepal_width`: 꽃받침의 넓이
- `petal_length`: 꽃잎의 길이
- `petal_width`: 꽃잎의 넓이

In [21]:
iris = sns.load_dataset('iris')
iris.head()

sepal_length  sepal_width  petal_length  petal_width species
0          5.10         3.50          1.40         0.20  setosa
1          4.90         3.00          1.40         0.20  setosa
2          4.70         3.20          1.30         0.20  setosa
3          4.60         3.10          1.50         0.20  setosa
4          5.00         3.60          1.40         0.20  setosa

- `sepal` 컬럼을 생성하고, `sepal_length` 와 `sepal_width`를 곱한 값을 대입하세요
- 상위 5개 행만 출력해 주세요

<p><strong>[출력 결과]</strong></p>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sepal_length</th>
      <th>sepal_width</th>
      <th>petal_length</th>
      <th>petal_width</th>
      <th>species</th>
      <th>sepal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>5.10</td>
      <td>3.50</td>
      <td>1.40</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>17.85</td>
    </tr>
    <tr>
      <th>1</th>
      <td>4.90</td>
      <td>3.00</td>
      <td>1.40</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>14.70</td>
    </tr>
    <tr>
      <th>2</th>
      <td>4.70</td>
      <td>3.20</td>
      <td>1.30</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>15.04</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4.60</td>
      <td>3.10</td>
      <td>1.50</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>14.26</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5.00</td>
      <td>3.60</td>
      <td>1.40</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>18.00</td>
    </tr>
  </tbody>
</table>
</div>

- `petal` 컬럼을 생성하고, `petal_length` 와 `petal_width`를 곱한 값을 대입하세요
- 상위 5개 행만 출력해 주세요

<p><strong>[출력 결과]</strong></p>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sepal_length</th>
      <th>sepal_width</th>
      <th>petal_length</th>
      <th>petal_width</th>
      <th>species</th>
      <th>sepal</th>
      <th>petal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>5.10</td>
      <td>3.50</td>
      <td>1.40</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>17.85</td>
      <td>0.28</td>
    </tr>
    <tr>
      <th>1</th>
      <td>4.90</td>
      <td>3.00</td>
      <td>1.40</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>14.70</td>
      <td>0.28</td>
    </tr>
    <tr>
      <th>2</th>
      <td>4.70</td>
      <td>3.20</td>
      <td>1.30</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>15.04</td>
      <td>0.26</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4.60</td>
      <td>3.10</td>
      <td>1.50</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>14.26</td>
      <td>0.30</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5.00</td>
      <td>3.60</td>
      <td>1.40</td>
      <td>0.20</td>
      <td>setosa</td>
      <td>18.00</td>
      <td>0.28</td>
    </tr>
  </tbody>
</table>
</div>

- `petal_length`, `petal_width` 컬럼을 제거하세요
- 상위 5개 행만 출력해 주세요

<p><strong>[출력 결과]</strong></p>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sepal_length</th>
      <th>sepal_width</th>
      <th>species</th>
      <th>sepal</th>
      <th>petal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>5.10</td>
      <td>3.50</td>
      <td>setosa</td>
      <td>17.85</td>
      <td>0.28</td>
    </tr>
    <tr>
      <th>1</th>
      <td>4.90</td>
      <td>3.00</td>
      <td>setosa</td>
      <td>14.70</td>
      <td>0.28</td>
    </tr>
    <tr>
      <th>2</th>
      <td>4.70</td>
      <td>3.20</td>
      <td>setosa</td>
      <td>15.04</td>
      <td>0.26</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4.60</td>
      <td>3.10</td>
      <td>setosa</td>
      <td>14.26</td>
      <td>0.30</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5.00</td>
      <td>3.60</td>
      <td>setosa</td>
      <td>18.00</td>
      <td>0.28</td>
    </tr>
  </tbody>
</table>
</div>

- `species`가 **setosa**인 꽃 중 `sepal`을 기준으로 내림차순 정렬 하세요.
- 상위 10개의 행만 출력하세요

<p><strong>[출력 결과]</strong></p>
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>sepal_length</th>
      <th>sepal_width</th>
      <th>species</th>
      <th>sepal</th>
      <th>petal</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>15</th>
      <td>5.70</td>
      <td>4.40</td>
      <td>setosa</td>
      <td>25.08</td>
      <td>0.60</td>
    </tr>
    <tr>
      <th>14</th>
      <td>5.80</td>
      <td>4.00</td>
      <td>setosa</td>
      <td>23.20</td>
      <td>0.24</td>
    </tr>
    <tr>
      <th>33</th>
      <td>5.50</td>
      <td>4.20</td>
      <td>setosa</td>
      <td>23.10</td>
      <td>0.28</td>
    </tr>
    <tr>
      <th>18</th>
      <td>5.70</td>
      <td>3.80</td>
      <td>setosa</td>
      <td>21.66</td>
      <td>0.51</td>
    </tr>
    <tr>
      <th>32</th>
      <td>5.20</td>
      <td>4.10</td>
      <td>setosa</td>
      <td>21.32</td>
      <td>0.15</td>
    </tr>
    <tr>
      <th>5</th>
      <td>5.40</td>
      <td>3.90</td>
      <td>setosa</td>
      <td>21.06</td>
      <td>0.68</td>
    </tr>
    <tr>
      <th>16</th>
      <td>5.40</td>
      <td>3.90</td>
      <td>setosa</td>
      <td>21.06</td>
      <td>0.52</td>
    </tr>
    <tr>
      <th>10</th>
      <td>5.40</td>
      <td>3.70</td>
      <td>setosa</td>
      <td>19.98</td>
      <td>0.30</td>
    </tr>
    <tr>
      <th>48</th>
      <td>5.30</td>
      <td>3.70</td>
      <td>setosa</td>
      <td>19.61</td>
      <td>0.30</td>
    </tr>
    <tr>
      <th>46</th>
      <td>5.10</td>
      <td>3.80</td>
      <td>setosa</td>
      <td>19.38</td>
      <td>0.32</td>
    </tr>
  </tbody>
</table>
</div>

`sepal` 컬럼의 평균과 `petal` 컬럼의 평균의 차이를 구하세요